rdr_service/resource/generators/consent_metrics.py

```diff
@@ -0,0 +1,254 @@
+        :param id_list: List of specific consent_file record IDs to retrieve.  Takes precedence over date_filter
+        :param date_filter:  A date string in YYYY-MM-DD format to use for filtering consent_file records. The
+                             default retrieves all records since consent validation started (in all environments)
+        :return:  A result set from the query of consent validation data
+        """
+        if not dao:
+            dao = self.ro_dao or ResourceDataDao(backup=True)
+
+        with dao.session() as session:
+            query = session.query(ConsentFile.id,
+                                  ConsentFile.created,
+                                  ConsentFile.modified,
+                                  ConsentFile.participant_id,
+                                  ConsentFile.type,
+                                  ConsentFile.sync_status,
+                                  ConsentFile.file_exists,
+                                  ConsentFile.is_signature_valid,
+                                  ConsentFile.is_signing_date_valid,
+                                  ConsentFile.other_errors,
+                                  # These ConsentFile fields are used to filter out false positives
+                                  ConsentFile.expected_sign_date,
+                                  ConsentFile.signing_date,
+                                  ParticipantSummary.dateOfBirth,
+                                  ParticipantSummary.consentForStudyEnrollmentFirstYesAuthored,
+                                  ParticipantSummary.consentForStudyEnrollmentAuthored,
+                                  ParticipantSummary.consentForCABoRAuthored,
+                                  ParticipantSummary.consentForElectronicHealthRecordsFirstYesAuthored,
+                                  ParticipantSummary.consentForElectronicHealthRecordsAuthored,
+                                  ParticipantSummary.consentForGenomicsRORAuthored,
+                                  HPO.hpoId,
+                                  HPO.name.label('hpo_name'),
+                                  Organization.organizationId,
+                                  Organization.displayName.label('organization_name'))\
+                  .join(ParticipantSummary, ParticipantSummary.participantId == ConsentFile.participant_id)\
+                  .outerjoin(HPO, HPO.hpoId == ParticipantSummary.hpoId)\
+                  .outerjoin(Organization, ParticipantSummary.organizationId == Organization.organizationId)
+
+            if id_list and len(id_list):
+                query = query.filter(ConsentFile.id.in_(id_list))
+            else:
+                query = query.filter(ConsentFile.modified >= date_filter)
+
+            # TODO:  Remove this vibrent filter once we're validating CE consents
+            results = query.filter(ParticipantSummary.participantOrigin == 'vibrent').all()
+            if not len(results):
```

Comment:
Since this is using an `.all()` which returns a list either way, probably better to use `if not results:`